demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -896,40 +900,47 @@ def validate_modeling_rule(
-            api_key=api_key,  # type: ignore[arg-type]
-            auth_id=auth_id,  # type: ignore[arg-type]
-            token=xsiam_token,  # type: ignore[arg-type]
-            collector_token=collector_token,  # type: ignore[arg-type]
-        )
-        xsiam_client = XsiamApiClient(xsiam_client_cfg)
-        if not verify_pack_exists_on_tenant(
-            xsiam_client, retrying_caller, modeling_rule, interactive
+        test_data = TestData.parse_file(modeling_rule.testdata_path.as_posix())
+        if (
+            Validations.TEST_DATA_CONFIG_IGNORE.value
+            not in test_data.ignored_validations
         ):
-            test_case = TestCase(
-                "Pack not installed on tenant", classname="Modeling Rule"
+            logger.info(
+                "[green]test data config is not ignored starting the test data validation...[/green]",
+                extra={"markup": True},
             )
-            return add_result_to_test_case(
-                "Pack not installed on tenant",
-                test_case,
-                modeling_rule_test_suite,
+            missing_event_data, _ = is_test_data_exists_on_server(
+                modeling_rule.testdata_path
             )
-        test_data = TestData.parse_file(modeling_rule.testdata_path.as_posix())
 
-        schema_test_case = TestCase(
-            "Validate Schema",
-            classname=f"Modeling Rule {get_relative_path_to_content(modeling_rule.schema_path)}",
-        )
-        if schema_path := modeling_rule.schema_path:
-            with open(modeling_rule.schema_path) as schema_file:
+            # initialize xsiam client
+            xsiam_client_cfg = XsiamApiClientConfig(
+                base_url=xsiam_url,  # type: ignore[arg-type]
+                api_key=api_key,  # type: ignore[arg-type]
+                auth_id=auth_id,  # type: ignore[arg-type]
+                token=xsiam_token,  # type: ignore[arg-type]
+                collector_token=collector_token,  # type: ignore[arg-type]
+            )
+            xsiam_client = XsiamApiClient(xsiam_client_cfg)
+            if not verify_pack_exists_on_tenant(
+                xsiam_client, retrying_caller, modeling_rule, interactive
+            ):
+                test_case = TestCase(
+                    "Pack not installed on tenant", classname="Modeling Rule"
+                )
+                return add_result_to_test_case(
+                    "Pack not installed on tenant",
+                    test_case,
+                    modeling_rule_test_suite,
+                )
+
+            schema_test_case = TestCase(
+                "Validate Schema",
+                classname=f"Modeling Rule {get_relative_path_to_content(modeling_rule.schema_path)}",
+            )
+            if schema_path := modeling_rule.schema_path:
                 try:
-                    schema = json.load(schema_file)
+                    schema = get_file(modeling_rule.schema_path)
```

Comment:
revert, it won't catch the json decoder error, but would return empty dict, it would cause us weird behaivor